crates/net/network/src/config.rs

```diff
@@ -467,6 +479,12 @@ impl NetworkConfigBuilder {
         self.build(Default::default())
     }
 
+    /// Sets the NAT resolver for external IP.
+    pub const fn add_nat(mut self, nat: Option<NatResolver>) -> Self {
```

Comment:
Added this mainly so that the unit test can be written more conveniently, would love to know if there's a better way